modules/mysql/mysql.go

```diff
@@ -0,0 +1,82 @@
+package mysql
+
+import (
+	"context"
+	"fmt"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+	"strings"
+)
+
+const rootUser = "root"
+const defaultUser = "test"
+const defaultPassword = "test"
+const defaultDatabaseName = "test"
+
+// MySQLContainer represents the MySQL container type used in the module
+type MySQLContainer struct {
+	testcontainers.Container
+	config *Config
+}
+
+// StartContainer creates an instance of the MySQL container type
+func StartContainer(ctx context.Context, image string, opts ...Option) (*MySQLContainer, error) {
```

Comment:
I think you could probably create a new Option for `WithImage`, that sets it in the config struct. That would simplify the method signature and we will be closer to the design we have in mind for an eventual refactor

Reply:
I was wondering about it because, for me at least and based on tc-java, it would be nice to declare the image to use when declaring `StartContainer` instead of relying on defaults. `Testcontainers for Java` used to have defaults in the past and now it promotes to always declare the image. Do you think this is something to take into account? We can also save to pass another option.

Reply:
Interesting feedback. Do you suggest that the StartContainer func should include a parameter for the image?

Reply:
in the current approach, yes. It also save to have an additional option 😄

Also, didn't mention it but in Testcontainers for Java an assertion is performed against the image name because the module works in this case with `mysql` but if you pass `my-custom-mysql` then we can not make sure it will work unless there is an option to say it is also compatible with mysql. This is already implemented in Java.

The way I see it, with this proposal we can be ready to introduce it smoothly.

Reply:
Indeed, it saves the option confusing the default but forces to add a validation on the image name, unless we let it fall